### PR TITLE
ci: add windows/macos to clippy to reproduce compilation error on windows 

### PR DESCRIPTION
**Pull request by delta1**
this PR adds windows and macos to the clippy job to reproduce the compilation failure in #186 

~~also fixes some clippy warns/errs~~

### CODE REVIEW

File: src/human_encoding/error.rs [hunk 1]
```diff
@@ -88,13 +88,13 @@ impl ErrorSet {
     pub fn add<P: Into<Position>, E: Into<Error>>(&mut self, position: P, err: E) {
         self.errors
             .entry(Some(position.into()))
-            .or_insert(vec![])
+            .or_default()
```
Comment (uncomputable) [inline src/human_encoding/error.rs:91]: 7202f933fddd92fc119835292d7f141de3d0d8c2: These changes were already merged (454727ea1e4d070987592469fa031454dcdeef02). Please rebase.

File: src/human_encoding/parse/mod.rs [hunk 2]
```diff
@@ -605,8 +605,8 @@ mod tests {
                 assert_eq!(main.cmr().to_string(), cmr);
 
                 let witness: HashMap<_, _> = witness
-                    .into_iter()
-                    .map(|(name, value)| (Arc::from(*name), Value::power_of_two(*value)))
+                    .iter()
```
Comment (uncomputable) [inline src/human_encoding/parse/mod.rs:608]: 7202f933fddd92fc119835292d7f141de3d0d8c2: For some reason this does not trigger a clippy lint in Rust 1.74.1 (Linux)

**Comment by delta1** [comment]
I will remove the clippy commit

**Comment by delta1** [comment]
closing for https://github.com/BlockstreamResearch/simplicity/pull/211

**Comment by delta1** [comment]
Is it worth having this change in CI to check macos/windows compilation?

**Comment by delta1** [comment]
rebased

**Comment by uncomputable** [comment]
Depends on #191. Let's merge this PR once CI runs.

**Comment by uncomputable** [comment]
@delta1 Feel free to rebase

**Comment by delta1** [comment]
@uncomputable rebased. windows ci failure is fixed upstream and in #192

**Comment by uncomputable** [comment]
Let's rebase on top of the merged #192 to fix CI.

**Comment by delta1** [comment]
@uncomputable done. macos failure looks spurious

**Comment by apoelstra** [comment]
@uncomputable this crate currently doesn't build with the `elements` feature and apparently our CI doesn't check this. It was broken in #189.

I can't run my local CI, or even `cargo test`, until this is fixed.

**Comment by uncomputable** [comment]
@apoelstra I thought our CI would catch this, but apparently it doesn't. `master` doesn't build on my machine :/

Fix `master` in a separate PR.

**Review by uncomputable — APPROVED** [review]
ACK 18012794a19442139e29ecb2e700c00cf0733562 This PR is a small change to CI. `master` is already broken.

**Pull request merged**